modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java

```diff
@@ -0,0 +1,318 @@
+ */
+public class PaginationBarTag extends BaseContainerTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		PaginationBarLabels labels = getLabels();
+
+		if (labels == null) {
+			setLabels(
+				new PaginationBarLabels(
+					"Showing {0} to {1} of {2}", "{0} items", "{0} items"));
+		}
+
+		return super.doStartTag();
+	}
+
+	public Integer getActiveDelta() {
+		return _activeDelta;
+	}
+
+	public Integer getActivePage() {
+		return _activePage;
+	}
+
+	public List<PaginationBarDelta> getDeltas() {
+		return _deltas;
+	}
+
+	public List<Integer> getDisabledPages() {
+		return _disabledPages;
+	}
+
+	public Integer getEllipsisBuffer() {
+		return _ellipsisBuffer;
+	}
+
+	public PaginationBarLabels getLabels() {
+		return _labels;
+	}
+
+	public Integer getTotalItems() {
+		return _totalItems;
+	}
+
+	public void setActiveDelta(Integer activeDelta) {
+		_activeDelta = activeDelta;
+	}
+
+	public void setActivePage(Integer activePage) {
+		_activePage = activePage;
+	}
+
+	public void setDeltas(List<PaginationBarDelta> deltas) {
+		_deltas = deltas;
+	}
+
+	public void setDisabledPages(List<Integer> disabledPages) {
+		System.out.println("Disabled pages set");
+		_disabledPages = disabledPages;
+	}
+
+	public void setEllipsisBuffer(Integer ellipsisBuffer) {
+		_ellipsisBuffer = ellipsisBuffer;
+	}
+
+	public void setLabels(PaginationBarLabels labels) {
+		_labels = labels;
+	}
+
+	public void setTotalItems(Integer totalItems) {
+		_totalItems = totalItems;
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_activeDelta = null;
+		_activePage = null;
+		_deltas = null;
+		_disabledPages = null;
+		_ellipsisBuffer = null;
+		_labels = null;
+		_totalItems = null;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/PaginationBar";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(Map<String, Object> props) {
+		props.put("activeDelta", _activeDelta);
+		props.put("activePage", _activePage);
+		props.put("deltas", _deltas);
+		props.put("disabledPages", _disabledPages);
+		props.put("ellipsisBuffer", _ellipsisBuffer);
+		props.put("labels", _labels);
+		props.put("totalItems", _totalItems);
+
+		return super.prepareProps(props);
+	}
+
+	@Override
+	protected int processStartTag() throws Exception {
+		super.processStartTag();
+
+		JspWriter jspWriter = pageContext.getOut();
+
+		jspWriter.write("<div class=\"pagination-bar\"><div class=\"");
+		jspWriter.write("dropdown pagination-items-per-page\">");
+
+		ResourceBundle resourceBundle = TagResourceBundleUtil.getResourceBundle(
+			pageContext);
+
+		Integer activeDelta = getActiveDelta();
+
+		if (activeDelta == null) {
+			PaginationBarDelta defaultDelta = getDeltas().get(0);
+
+			Integer label = (Integer)defaultDelta.get("label");
+
+			activeDelta = label;
+		}
+
+		jspWriter.write("<button class=\"dropdown-toggle btn btn-unstyled\"");
```

Comment:
A taglib newbie question: Could We reuse clay:button in someway instead of using button html str? It will be good to get the render string from a clay:button and reuse it :(

Reply:
No, in this case we can't because `clay:button` renders icon to the "left" of the label, in this case we need it on the right

Reply:
Great, I understand now that it's inapplicable to this work but there is a way to use a tag in another just using Java? *Without using jspWriter

Reply:
No, unfortunately ... but you know `ButtonTag` and many others use `JspWriter` under the hood, I don't see any problem with that, this isn't React.